### PR TITLE
Lemon Quality Segmentation #216

### PR DESCRIPTION
**Pull request by aditya0929**
# Pull Request for DL-Simplified 💡

## Issue Title : Lemon Quality Segmentation #216

- **Info about the related issue (Aim of the project)** : Lemon image classification between good and bad quality
- **Name:** Aditya Narayan Jha
- **GitHub ID:** https://github.com/aditya0929
- **Email ID:** [adityajha8906@gmail.com](mailto:adityajha8906@gmail.com)
- **Idenitfy yourself:** `SSOC-2023 Participant`

Closes: #216 

### Describe the add-ons or changes you've made 📃

I chose convolutional neural networks for the multi instance classification situation since the dataset included contained empty-background photos, which lacked any lemons. Even though I had thought of going for binary classification in the start . 

First, I tried training the data across a variety of numbers of epochs until a satisfactory level of accuracy was achieved using a basic CNN with maxpooling and the addition of additional layers as needed.

I used the same strategy to train Alexnet and VGG16, two additional models.

I have also included the learning tools that were very helpful to me in developing the model and that include a lot more in-depth information.

In the models folder , i have added comparioson.ipynb file which containes different graphical representation to compare between accuracy percentages of the models 

## Type of change ☑️

What sort of change have you made:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

To keep a clean code and concentrate on one model at a time, I decided to take the approach of creating separate ipynb files for each models.  Aside from the model architecture, all models' code adheres to the same process for all other aspects. 

I tested it by first hiding the actual labels and then displaying the predicted labels, which turned out to be always accurate thanks to the high accuracy rates of all three models.

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

 **P.S.** - Thank you so much @abhisheks008 for this opportunity .  

I had started working on the models from the day i had commented on the issue .

 I am attaching accuracy score of all three modules for references and predicted labels . 

predicted labels
Accuracy Evaluation

### CODE REVIEW

**Comment by aditya0929** [comment]
Hi , @abhisheks008  I have tried to make the requested changes . Please take a look into it . 
THANK YOU .

**Review by abhisheks008 — APPROVED** [review]
You have done a great job. Pull request approved and ready to be merged.

Keep contributing in open source!!
@aditya0929
